backend/services/deviceauth/server.go

```diff
@@ -121,13 +121,12 @@ func RunServer(c config.Reader) error {
 
 		devauth = devauth.WithCache(cache)
 	}
-
-	apiHandler := api_http.NewRouter(devauth, db)
-
-	if err != nil {
-		return errors.Wrap(err, "device authentication API handlers setup failed")
+	options := []api_http.Option{
+		api_http.SetMaxRequestSize(int64(c.GetInt(dconfig.SettingMaxRequestSize))),
 	}
 
+	apiHandler := api_http.NewRouter(devauth, db, options...)
```

Comment:
No need to create the slice.
```suggestion
	apiHandler := api_http.NewRouter(devauth, db, api_http.SetMaxRequestSize(int64(c.GetInt(dconfig.SettingMaxRequestSize))))
```